lib/ProductOpener/Tags.pm

```diff
@@ -4613,6 +4613,49 @@ sub add_tags_to_field ($product_ref, $tag_lc, $field, $additional_fields) {
 	return;
 }
 
+sub remove_tag_from_field($product_ref, $tag_lc, $field, $tags_to_remove) {
+
+	my %existing = ();
+	if (defined $product_ref->{$field . "_tags"}) {
+		foreach my $tagid (@{$product_ref->{$field . "_tags"}}) {
+			$existing{$tagid} = 1;
+		}
+	}
+
+	my @removed_tags = ();
+	my @tags_to_remove = split(/,/, $tags_to_remove);
+
+	foreach my $tag (@tags_to_remove) {
+
+		$tag =~ s/^\s+//;
+		$tag =~ s/\s+$//;
+
+		my $tagid;
+
+		if (defined $taxonomy_fields{$field}) {
+			$tagid = get_taxonomyid($tag_lc, canonicalize_taxonomy_tag($tag_lc, $field, $tag));
+		}
+		else {
+			$tagid = get_string_id_for_lang($tag_lc, $tag);
+		}
+		if (exists $existing{$tagid}) {
+			delete $existing{$tagid};
+			push @removed_tags, $tag;
+		}
+	}
```

Comment:
Instead of this I propose this implementation to you:
* create a hashmap of tags to remove (as you did with existing)
* get the list of tags (and keep it's length for reference)
* use [grep](https://perldoc.perl.org/functions/grep) to filter the list of tags
* update the field if the list length as changed

Reply:
Sorry ignore this comment, I read a bit quickly and ignored some of the complexity that you instead handled well !
Your implementation is better.